### PR TITLE
Add Alisa submitter

### PR DESCRIPTION
**Pull request by Yancey0623**
part of #1321 

This PR import `goalisa` and add `alisaSubmitter` struct.

Before I continue to implement `alisaSubmitter`, I would like to clean up `submitter` code,  move the existing submitter code to `pkg/submitter`.

### CODE REVIEW

**Comment by wangkuiyi** [comment]
Thanks for this scaffolding PR. It expresses the idea clearly. Could you please take a look at my questions in https://github.com/sql-machine-learning/sqlflow/pull/1610#pullrequestreview-337713570?

File: pkg/sql/submitter.go [hunk 1]
```diff
@@ -31,15 +31,25 @@ import (
...
 
-var envSubmitter = os.Getenv("SQLFLOW_submitter")
-
 var submitterRegistry = map[string](Submitter){
 	"default": &defaultSubmitter{},
 	// TODO(typhoonzero): add submitters like alps, elasticdl
 }
 
+// SubmitterRegister registes a submitter
+func SubmitterRegister(name string, submitter Submitter) {
```
Comment (typhoonzero) [inline pkg/sql/submitter.go:40]: Use a static map and remove `SubmitterRegister`, `pai_submitter.go:init()` should be simpler I think.

**Review by typhoonzero — COMMENTED** [review]
@wangkuiyi maybe we can merge this to implement alisa job submitting and continue refactoring according to: https://github.com/sql-machine-learning/sqlflow/issues/1585

**Comment by wangkuiyi** [comment]
@typhoonzero I see. This PR doesn't depend on #1610 . I am OK to merge it and clean it up later.

**Review by wangkuiyi — APPROVED** [review]

File: pkg/sql/submitter.go (hunk 1 above)
Comment (Yancey0623) [reply pkg/sql/submitter.go:40]: It just used to be consistent with the current implementation suck as `pai_submitter.go`, it used `init()` to register the submitter.   I would clean up the submitters in #1610 .

**Pull request merged**